internal/mode/static/telemetry/collector.go

```diff
@@ -147,3 +160,16 @@ func collectGraphResourceCount(
 
 	return ngfResourceCounts, nil
 }
+
+func (c DataCollectorImpl) collectClusterID(ctx context.Context) (string, error) {
+	kubeSystem := types.NamespacedName{
+		Name: KubeSystem,
+	}
+	var nsMetadata v1.Namespace
+	err := c.cfg.K8sClientReader.Get(ctx, kubeSystem, &nsMetadata)
+	if err != nil {
+		return "", fmt.Errorf("failed to get metadata for cluster :%w", err)
```

Comment:
Even though we are only interested in the metadata of the namespace, we are actually getting the entire namespace here. So I would change the error message to:
```suggestion
		return "", fmt.Errorf("failed to get namespace %q :%w", KubeSystem, err)
```